app/app/views.py

```diff
@@ -24,9 +24,34 @@ def institutions(request):
     context = {}
 
     institutions = Institution.objects.annotate(project_count=Count("project")).order_by("name")
+
+    institutions_array = []
+
+    for institution in institutions:
+        institution_dict = {
+            "name": institution.name,
+            "abbreviation": institution.abbreviation,
+            "url": institution.url,
+            "email": institution.email,
+            "logo": institution.logo,
+        }
+        empty_value_count_cal = sum(1 for value in institution_dict.values() if value == "")
+
+        """Rating returns % number of empty fields.
+
+        Profile completion weighting is % number of completed fields in 5 star rating.
+
+        Profile completion is calculated using the number of present model fields,
+        model fields have to be added to the institution_dict"""
+
+        rating = (empty_value_count_cal / len(institution_dict)) * 100
+        institution_dict["empty_fields"] = round(rating)
```

Comment:
From here on, your code calls the rating "empty_fields", which is a bit inconsistent. I'd say, let's hide the "empty fields" implementation detail here, and only call it "rating" in the context and template. In future the rating might be calculated a different way, and not (only) be about empty fields.

But then, it is a bit odd that a full (good) profile gets a 0 rating, which would usually convey a bad rating. Can I suggest that a rating of 100 is a "good" profile? If it helps in the implementation, you can rather count provided fields, i.e. `completed_fields = sum(1 for value in institution_dict.values() if value)`, but see what works well for you.

Reply:
Okay, I will change and do it the other way around